x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/actions_panel/actions_panel.tsx

```diff
@@ -54,38 +48,20 @@ export const ActionsPanel: FC<Props> = ({
     const indexPatternId = indexPattern.id;
     const getDiscoverUrl = async (): Promise<void> => {
       const isDiscoverAvailable = capabilities.discover?.show ?? false;
-      if (!isDiscoverAvailable) {
-        return;
-      }
-
-      const state: DiscoverUrlGeneratorState = {
+      if (!isDiscoverAvailable) return;
+      if (!discover.locator) return;
```

Comment:
Would be good to throw an error message/show it in console indicating that Discover locator is not available for easy troubleshooting